### ISSUE TITLE
Extension is always loaded - would be worthwhile to review `activationEvents`

### ISSUE THREAD

**Issue by robole** [chriskirknielsen/visubezier#8]
Hi Chris,

I like this addition to VS Code! It was my first time dabbling with `cubic-bezier()` and I found this!

One thing that is worth changing is the activation event that triggers the loading of the extension. Currently the extension loads every time VS Code is opened, and with the asterisk activation event, it effectively delays VS Code from becoming active. The VS Code docs gives the [following advice on this](https://code.visualstudio.com/api/references/activation-events#Start-up):

> To ensure a great end user experience, please use this activation event in your extension only when no other activation events combination works in your use-case.

I would consider targeting CSS syntaxes instead. You can narrow down the languages you are targeting by using the `onLanaguage` event instead. You can use something like this instead:

"activationEvents": [
    "onLanguage:css",
    "onLanguage:scss",
  ],

The scss [language ID](https://code.visualstudio.com/docs/languages/identifiers) includes .sass files. You could include single file component formats such as Vue and Svelte also too.

If you have some difficulty in deciding on which languages to target, you could use `onStartUpFinished` instead which would remove it from blocking VS Code on startup at least.

Thanks.

**Comment by chriskirknielsen** [comment]
Hey Rob!

Thank you for filing this. I can definitely see where you're coming from. I had made an assumption, that if you download this plugin, you work frequently on front-end stuff, meaning that you'll likely always have a filetype that could carry an easing value. That might be an oversimplification, so maybe we can make the change you suggested.

However, while CSS and Sass are the main targets, you can have inline styles with an easing (less likely but not impossible), a hardcoded easing in JS, an easing set as a design token in a JSON file, HTML-adjencent formats like SVG and any templating language you can think of, and like you mentioned, single-file components.

So ultimately, that would leave us with quite a list of formats, which isn't too hard to list out, but given the number of possibilities, it might exclude somebody's workflow. Which is why I initially set `*` to avoid excluding people.

With that in mind, I'd gladly set up individual activation events, but I'd first need to make sure we cover most cases.

```json
"activationEvents": [
	"onLanguage:css",
	"onLanguage:scss",
	"onLanguage:less",
	"onLanguage:stylus",
	"onLanguage:html",
	"onLanguage:svg",
	"onLanguage:json",
	"onLanguage:jsonc",
	"onLanguage:javascript",
	"onLanguage:javascriptreact",
	"onLanguage:typescript",
	"onLanguage:typescriptreact",
	"onLanguage:vue",
	"onLanguage:vue-html",
	"onLanguage:svelte",
	"onLanguage:astro",
	"onLanguage:nunjucks",
	"onLanguage:handlebars",
	"onLanguage:haml",
	"onLanguage:slim",
	"onLanguage:jade",
	"onLanguage:php",
	"onLanguage:razor",
],
```

Does that seem like an exhaustive enough list?

I'll also have a look at the VS Code documentation: perhaps users can add a language as an extension parameter to cover their own needs, which might reduce the list above.

Cheers!

**Comment by robole** [comment]
Hi Chris,

Thanks for the quick response!

The list looks good. It gets quite long when you want to cover the bases to leave it open like that!

If you want to leave the door totally open for people to specify their own languages, you could :
- use `onStartUpFinished` activation event
- Add a workspace setting  to the package.json that will represent an array of languages
- In JS, in the `activate()` method you can consult your array of languages  to decide to add the decorators or not. 

It's up to you which route you take. The most important thing is not to block startup of VS Code. It is preferable not to have the extension loaded always, but the benefits are probably only realized by folks who work on backend projects.

I use VS Code for Java and a few other backend languages, you can slow VS Code down quite a bit if you have a lot of extensions that are loaded immediately like this. VS Code is not great for polyglot development in that respect.

**Comment by tjx666** [comment]
Maybe ref https://github.com/tjx666/vscode-colorize/blob/develop/package.json

**Comment by chriskirknielsen** [comment]
This should now be solved with [v1.5.0](https://github.com/chriskirknielsen/visubezier/releases/tag/v1.5.0). Thanks for the proposal, @robole, and thanks @tjx666 for the suggested list of languages! I am not very well versed in VSCode Extensions so making this work with other methods is a bit obscure to me at the moment, hence why I went with the third option you listed, Rob.

If this feels like the incorrect way to make this work, I'm happy to accept PRs. 😄
